R/population-simulation-task.R

```diff
@@ -0,0 +1,81 @@
+#' @title SimulationTask
+#' @description  R6 class for SimulationTask settings
+#' @field numberOfCores number of cores for parallel computation
+#' @field getTaskResults function called by task that computes and format figure results
+PopulationSimulationTask <- R6::R6Class(
+  "PopulationSimulationTask",
+  inherit = SimulationTask,
+  public = list(
+    numberOfCores = NULL,
+    getTaskResults = NULL,
+
+    #' @description
+    #' Create a `SimulationTask` object
+    #' @param numberOfCores number of cores for parallel computation
+    #' @param getTaskResults function called by task that computes and format figure results
+    #' @param ... parameters inherited from R6 class `Task` object
+    #' @return A new `SimulationTask` object
+    initialize = function(numberOfCores = NULL,
+                          getTaskResults = NULL,
+                          ...) {
+      super$initialize(...)
+      self$updateNumberOfCores(numberOfCores %||% defaultSimulationNumberOfCores)
+      self$getTaskResults <- getTaskResults
+    },
+
+    #' @description
+    #' Update the `numberOfCores`
+    #' @param numberOfCores is the number of cores to use for simulation
+    updateNumberOfCores = function(numberOfCores) {
+      if (!is.null(numberOfCores)) {
+        validateIsInteger(numberOfCores)
+        validateIsOfLength(object = numberOfCores, nbElements = 1)
+        self$numberOfCores <- numberOfCores
+      }
+    },
+
+    #' @description
+    #' Save results from task run.
+    #' @param set R6 class `SimulationStructure`
+    #' @param taskResults list of results from task run.
+    saveResults = function(set,
+                           taskResults) {
+      ospsuite::exportResultsToCSV(
+        taskResults,
+        set$simulationResultFileNames
+      )
+    },
+
+    #' @description
+    #' Run task and save its output
+    #' @param structureSets list of `SimulationStructure` R6 class
+    runTask = function(structureSets) {
+      logWorkflow(
+        message = paste0("Starting ", self$message),
+        pathFolder = self$workflowFolder
+      )
+
+      if (!is.null(self$outputFolder)) {
+        dir.create(file.path(self$workflowFolder, self$outputFolder))
+      }
+
+      for (set in structureSets) {
+        logWorkflow(
+          message = paste0("Run simulation: ", set$simulationSet$simulationName),
+          pathFolder = self$workflowFolder
+        )
+        if (self$validateInput()) {
+          taskResults <- self$getTaskResults(
```

Comment:
This indentation looks kind of weird. Is this how the styler formatted the code?

Reply:
fixed

Reply:
Sorry if I was not clear. I meant the fact that  you write
```

          self$saveResults(
            set,
            taskResults
          )
``` 

instead of 
```
          self$saveResults(set,taskResults)
```

The former is not consistent with most code we have in the reporting engine